flink-core/src/main/java/org/apache/flink/configuration/ExecutionOptions.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.configuration;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+/**
+ * {@link ConfigOption}s specific for a single execution of a user program.
+ */
+@PublicEvolving
+public class ExecutionOptions {
+
+	/**
+	 * Should be moved to {@code ExecutionCheckpointingOptions} along with
```

Comment:
The answer is modules hierarchy.

This option is used in `ExecutionConfig` which is in flink-core module, whereas the `ExecutionCheckpointingOptions` should be in the `flink-streaming-java` module alongside `CheckpointConfig` (because e.g. they define an enum option for `ExternalizedCheckpointCleanup`).

Because of that structure I cannot put the `SNAPSHOT_COMPRESSION` option in `ExecutionCheckpointingOptions` because it would be inaccessible in `ExecutionConfig`. And on the other hand I cannot move the corresponding field from `ExecutionConfig` to `CheckpointConfig` because it would be a breaking change.